### ISSUE TITLE
No longer downloading from PIA

### ISSUE THREAD

**Issue by coulterj** [haugene/docker-transmission-openvpn#1594]
I think I started having the issues with connection to PIA about 5 days ago but not exactly sure.  Yesterday I was getting error messages in the log.  Today I am not getting errors that I can see.

Attaching to transmission-vpn
transmission-vpn    | Starting container with revision: 563c2e276737eb6bb3083793ecc7324b23b39327
transmission-vpn    | Creating TUN device /dev/net/tun
transmission-vpn    | mknod: /dev/net/tun: File exists
transmission-vpn    | Using OpenVPN provider: PIA
transmission-vpn    | Provider PIA has a custom setup script, executing it
transmission-vpn    | Downloading OpenVPN config bundle openvpn into temporary file /tmp/tmp.bFeooa
transmission-vpn    | Extract OpenVPN config bundle into PIA directory /etc/openvpn/pia
transmission-vpn    | Starting OpenVPN using config us_texas.ovpn
transmission-vpn    | Modifying /etc/openvpn/pia/us_texas.ovpn for best behaviour in this container
transmission-vpn    | Setting OpenVPN credentials...
transmission-vpn    | adding route to local network 192.168.86.0/24 via 172.21.0.1 dev eth0
transmission-vpn    | Mon Dec  7 08:07:45 2020 OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
transmission-vpn    | Mon Dec  7 08:07:45 2020 library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
transmission-vpn    | Mon Dec  7 08:07:45 2020 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
transmission-vpn    | Mon Dec  7 08:07:45 2020 CRL: loaded 1 CRLs from file [[INLINE]]
transmission-vpn    | Mon Dec  7 08:07:46 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]154.29.131.72:1198
transmission-vpn    | Mon Dec  7 08:07:46 2020 UDP link local: (not bound)
transmission-vpn    | Mon Dec  7 08:07:46 2020 UDP link remote: [AF_INET]154.29.131.72:1198
transmission-vpn    | Mon Dec  7 08:07:46 2020 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
transmission-vpn    | Mon Dec  7 08:07:46 2020 [dallas411] Peer Connection Initiated with [AF_INET]154.29.131.72:1198
transmission-vpn    | Mon Dec  7 08:07:47 2020 OpenVPN ROUTE6: OpenVPN needs a gateway parameter for a --route-ipv6 option and no default was specified by either --route-ipv6-gateway or --ifconfig-ipv6 options
transmission-vpn    | Mon Dec  7 08:07:47 2020 OpenVPN ROUTE: failed to parse/resolve route for host/network: 2000::/3
transmission-vpn    | Mon Dec  7 08:07:47 2020 TUN/TAP device tun0 opened
transmission-vpn    | Mon Dec  7 08:07:47 2020 /sbin/ip link set dev tun0 up mtu 1500
transmission-vpn    | Mon Dec  7 08:07:47 2020 /sbin/ip addr add dev tun0 10.22.112.206/24 broadcast 10.22.112.255
transmission-vpn    | Mon Dec  7 08:07:47 2020 /etc/openvpn/tunnelUp.sh tun0 1500 1553 10.22.112.206 255.255.255.0 init
transmission-vpn    | Up script executed with tun0 1500 1553 10.22.112.206 255.255.255.0 init
transmission-vpn    | Updating TRANSMISSION_BIND_ADDRESS_IPV4 to the ip of tun0 : 10.22.112.206
transmission-vpn    | Updating Transmission settings.json with values from env variables
transmission-vpn    | Using existing settings.json for Transmission /data/transmission-home/settings.json
transmission-vpn    | Overriding bind-address-ipv4 because TRANSMISSION_BIND_ADDRESS_IPV4 is set to 10.22.112.206
transmission-vpn    | Overriding cache-size-mb because TRANSMISSION_CACHE_SIZE_MB is set to 16
transmission-vpn    | Overriding download-dir because TRANSMISSION_DOWNLOAD_DIR is set to /data/completed
transmission-vpn    | Overriding idle-seeding-limit because TRANSMISSION_IDLE_SEEDING_LIMIT is set to 1
transmission-vpn    | Overriding idle-seeding-limit-enabled because TRANSMISSION_IDLE_SEEDING_LIMIT_ENABLED is set to true
transmission-vpn    | Overriding incomplete-dir because TRANSMISSION_INCOMPLETE_DIR is set to /data/incomplete
transmission-vpn    | Overriding queue-stalled-enabled because TRANSMISSION_QUEUE_STALLED_ENABLED is set to true
transmission-vpn    | Overriding ratio-limit because TRANSMISSION_RATIO_LIMIT is set to 0.01
transmission-vpn    | Overriding ratio-limit-enabled because TRANSMISSION_RATIO_LIMIT_ENABLED is set to true
transmission-vpn    | Overriding rpc-authentication-required because TRANSMISSION_RPC_AUTHENTICATION_REQUIRED is set to true
transmission-vpn    | Overriding rpc-host-whitelist because TRANSMISSION_RPC_HOST_WHITELIST is set to 127.0.0.1,192.168.*.*
transmission-vpn    | Overriding rpc-password because TRANSMISSION_RPC_PASSWORD is set to [REDACTED]
transmission-vpn    | Overriding rpc-port because TRANSMISSION_RPC_PORT is set to 9091
transmission-vpn    | Overriding rpc-username because TRANSMISSION_RPC_USERNAME is set to admin
transmission-vpn    | Overriding umask because TRANSMISSION_UMASK is set to 2
transmission-vpn    | Overriding utp-enabled because TRANSMISSION_UTP_ENABLED is set to false
transmission-vpn    | Overriding watch-dir because TRANSMISSION_WATCH_DIR is set to /data/watch
transmission-vpn    | sed'ing True to true
transmission-vpn    | Enforcing ownership on transmission config directories
transmission-vpn    | Applying permissions to transmission config directories
transmission-vpn    | Setting owner for transmission paths to 1000:998
transmission-vpn    | Setting permission for files (644) and directories (755)
transmission-vpn    | Setting permission for watch directory (775) and its files (664)
transmission-vpn    |
transmission-vpn    | -------------------------------------
transmission-vpn    | Transmission will run as
transmission-vpn    | -------------------------------------
transmission-vpn    | User name:   abc
transmission-vpn    | User uid:    1000
transmission-vpn    | User gid:    998
transmission-vpn    | -------------------------------------
transmission-vpn    |
transmission-vpn    | STARTING TRANSMISSION
transmission-vpn    | Transmission startup script complete.
transmission-vpn    | Mon Dec  7 08:07:48 2020 WARNING: OpenVPN was configured to add an IPv6 route over tun0. However, no IPv6 has been configured for this interface, therefore the route installation may fail or may not work as expected.
transmission-vpn    | Mon Dec  7 08:07:48 2020 Initialization Sequence Completed

--
My YAML looks like

  transmission-vpn:
    container_name: transmission-vpn
    image: haugene/transmission-openvpn
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun
    restart: always
    ports:
    - "9091:9091"
    dns:
      - 1.1.1.1
      - 8.8.8.8
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ${USERDIR}/docker/transmission-vpn:/data
      - ${USERDIR}/docker/shared:/shared
      - /mnt/usb:/media
      - /mnt/usb/Downloads/Transmission/watch:/data/watch
      - /mnt/usb/Downloads/Transmission/completed:/data/completed
      - /mnt/usb/Downloads/Transmission/incomplete:/data/incomplete
    environment:
      - DISABLE_PORT_UPDATER=false
      - OPENVPN_PROVIDER=PIA
      - OPENVPN_USERNAME=XXXXXX
      - OPENVPN_PASSWORD=XXXXXX
      - PIA_OPENVPN_CONFIG_BUNDLE=openvpn
      - OPENVPN_CONFIG=us_texas
      - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
      - LOCAL_NETWORK=192.168.86.0/24
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TZ}
      - TRANSMISSION_RPC_AUTHENTICATION_REQUIRED=true
      - TRANSMISSION_RPC_HOST_WHITELIST="127.0.0.1,192.168.*.*"
      - TRANSMISSION_RPC_PASSWORD=password
      - TRANSMISSION_RPC_USERNAME=admin
      - TRANSMISSION_UMASK=2
      - TRANSMISSION_RATIO_LIMIT=0.01
      - TRANSMISSION_RATIO_LIMIT_ENABLED=true
      - TRANSMISSION_UTP_ENABLED=false
      - TRANSMISSION_QUEUE_STALLED_ENABLED=true
      - TRANSMISSION_IDLE_SEEDING_LIMIT_ENABLED=true
      - TRANSMISSION_IDLE_SEEDING_LIMIT=1
      - TRANSMISSION_CACHE_SIZE_MB=16

Basically, it no longer connects to the tracker and I am getting errors like this post which maybe related - https://github.com/haugene/docker-transmission-openvpn/issues/1584#issuecomment-739729363

I am able to access the the internet from the container

docker exec -it transmission-vpn ping google.com
PING google.com (142.250.68.174): 56 data bytes
64 bytes from 142.250.68.174: seq=0 ttl=117 time=13.779 ms
64 bytes from 142.250.68.174: seq=1 ttl=117 time=25.915 ms
64 bytes from 142.250.68.174: seq=2 ttl=117 time=16.616 ms
64 bytes from 142.250.68.174: seq=3 ttl=117 time=16.821 ms
64 bytes from 142.250.68.174: seq=4 ttl=117 time=16.936 ms

and am getting an IP from PIA
docker exec -it transmission-vpn wget -qO- http://ipecho.net/plain | xargs echo
154.3.250.78

**Comment by clement-z** [comment]
Did you try downloading something using different trackers? For example downloading [a release of Ubuntu](https://ubuntu.com/download/alternative-downloads)?

**Comment by coulterj** [comment]
That does seem to work...  however downloading from different trackers does not, multiple sites.  Any ideas?

![image](https://user-images.githubusercontent.com/19748538/101406265-d84f5c80-389e-11eb-8c4e-516f82c132ae.png)

**Comment by ilike2burnthing** [comment]
Try changing TRANSMISSION_RATIO_LIMIT to an integer (e.g. 1). Other possible solutions here - https://github.com/haugene/docker-transmission-openvpn/issues/1507

**Comment by coulterj** [comment]
Thanks for the link.  The remedy was to delete the transmission_home dir and let it recreate.
